dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -46,61 +44,59 @@
 
     @Override
     public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
-        SettableFuture<Response> future = SettableFuture.create();
-
-        IntFunction<ListenableFuture<Response>> callSupplier = attempt -> {
```

Comment:
Replaced this abstraction by passing the channel, endpoint, and request into the retryer. We still have the attempt  index available for instrumentation (RetryingCallback.failures), but I find it's easier to spot bugs without the indirection.